### PR TITLE
Change ref handling in TextualBody to prevent it parsing generated nodes

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/11627
Instead of passing in the entire event span and only looking at its children, only look at the event body and that way we don't end up re-parsing the url previews we created already (when an event edit comes in to update the same tile)

Removes unused paths for getInnerText

Needs further testing around:
- [x] Spoilers
- [x] Linkification
- [x] Pillification

I will write unit tests to support this.

Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### CODE REVIEW

File: src/async-components/views/dialogs/keybackup/CreateKeyBackupDialog.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ import MatrixClientPeg from '../../../../MatrixClientPeg';
 import { scorePassword } from '../../../../utils/PasswordScorer';
 import { _t } from '../../../../languageHandler';
 import { accessSecretStorage } from '../../../../CrossSigningManager';
-import SettingsStore from '../../../../../lib/settings/SettingsStore';
+import SettingsStore from '../../../../settings/SettingsStore';
```
Comment (turt2live) [inline src/async-components/views/dialogs/keybackup/CreateKeyBackupDialog.js]: thank you :D

File: src/components/views/context_menus/MessageContextMenu.js [hunk 2]
```diff
@@ -422,7 +422,7 @@ module.exports = createReactClass({
             </MenuItem>
         );
 
-        if (this.props.eventTileOps && this.props.eventTileOps.getInnerText) {
+        if (this.props.eventTileOps) { // this event is rendered using TextuaLBody
```
Comment (turt2live) [inline src/components/views/context_menus/MessageContextMenu.js]:
```suggestion
        if (this.props.eventTileOps) { // this event is rendered using TextualBody
```

File: src/components/views/elements/Pill.js [hunk 3]
```diff
@@ -261,7 +261,7 @@ const Pill = createReactClass({
         }
 
         const classes = classNames("mx_Pill", pillClass, {
-            "mx_UserPill_me": userId === MatrixClientPeg.get().credentials.userId,
+            "mx_UserPill_me": userId === MatrixClientPeg.get().getUserId(),
```
Comment (turt2live) [inline src/components/views/elements/Pill.js]: This is the second unrelated change in this PR - please try to keep these out for future PRs. Personally I don't mind having one vaguely related thing in the PR, but after a couple it starts to feel like false advertising on the PR.

Comment (t3chguy) [reply src/components/views/elements/Pill.js]: The change is not unrelated, it was to simplify stubbing for the test, any and all changes in this PR are related either to the change or testing it.

https://github.com/matrix-org/matrix-react-sdk/pull/3711/files#diff-c2d1eb236f1e50980bd094446ee76c63R132

File: src/async-components/views/dialogs/keybackup/CreateKeyBackupDialog.js (hunk 1 above)
Comment (t3chguy) [reply src/async-components/views/dialogs/keybackup/CreateKeyBackupDialog.js]: it broke my tests so :D

File: src/components/views/elements/Pill.js (hunk 3 above)
Comment (turt2live) [reply src/components/views/elements/Pill.js]: things like this should be pointed out so tired eyes don't miss them :(

**Pull request merged**